torch/ao/quantization/fx/prepare.py

```diff
@@ -1129,38 +1200,136 @@ def insert_observers_for_model(
     #
     cache_for_no_tensor_check: Dict[Node, bool] = {}
 
-    inputs_seen_counter = 0
-    outputs_seen_counter = 0
-
     # first, populate the dtype map based only on qconfig and qhandler
     # this assumes:
     # graph inputs are fp32 by default, and int8 where overriden
     # other nodes output dtype is specified by the qconfig
     named_modules = dict(model.named_modules(remove_duplicate=False))
+
+    input_quantized_idxs: List[int] = prepare_custom_config.input_quantized_indexes
+    output_quantized_idxs: List[int] = prepare_custom_config.output_quantized_indexes
+    processed_nodes: Set[Node] = set()
+    # initalize target_dtype_info
+    DEFAULT_QCONFIG_FOR_TARGET_DTYPE_INFO = (
+        QConfig(
+            activation=PlaceholderObserver.with_args(dtype=torch.float32),
+            weight=PlaceholderObserver.with_args(dtype=torch.float32)
+        )
+    )
+    for node in model.graph.nodes:
+        node.meta["target_dtype_info"] = {
+            "input_act_obs_or_fq_ctr": DEFAULT_QCONFIG_FOR_TARGET_DTYPE_INFO.activation,
+            "output_act_obs_or_fq_ctr": DEFAULT_QCONFIG_FOR_TARGET_DTYPE_INFO.activation
+        }
+
+    inputs_seen_counter = 0
+    outputs_seen_counter = 0
+    placeholder_node_to_input_index: Dict[Node, int] = {}
+    # TODO: we probably don't need this counter since each graph will only have
```

Comment:
the original plan was to implement anything a user could do, such as 

```
UserModule(foo, (bar, baz), {'foo': bar}, ...)
```

or 

```
return 1, (2, 3), {4: {5: 6}}
```

all or that was supported by PyTorch and FX symbolic trace, not sure if the story is different in PT2.0 export?

Reply:
my understanding is that for input we will have multiple placeholder nodes, but for output we only have one node
```
return 1, (2, 3), {4: {5: 6}}
```
will be one output node with an argument of a tuple(1, (2, 3), {4: {5: 6}}), also cc @qihqi to confirm as well

Reply:
yeah.  The counter is for the individual parts of the outputs, it shouldn't matter that it's a single node at the end.  For example, if it's `return (a, b)`, the user should be able to configure the dtype of `b`.

Reply:
yeah agree, we just don't have code to handle this right now